arbitrator/src/capabilities_interface.cpp

```diff
@@ -54,39 +55,56 @@ namespace arbitrator
                     // NOTE: It is assumed for now (due to lack of true capabilties register),
                     // That all plugins conform to the convention of providing a service for
                     // planning maneuvers at "/plugins/{PLUGIN_NAME}/strategic_plan/plan_maneuvers"
-                    std::string topic = "/guidance/plugins/" + it->name + "/" + capability;
+                    // std::string topic = "/guidance/plugins/" + it->name + "/" + capability;
                     // Check if bucket for that capability already exists
-                    if (capabilities_.find(capability) != capabilities_.end())
+                    if (capabilities_.find(capability) == capabilities_.end())
                     {
-                        capabilities_[capability].push_back(topic);
-                    }
-                    else
-                    {
-                        capabilities_.emplace(capability, std::vector<std::string>{topic});
+                        capabilities_.insert(capability);
                     }
 
                     // TODO: Special cased for now due to lack of broader capabilities service
-                    if (capability == STRATEGIC_PLAN_CAPABILITY)
-                    {
-                        ros::ServiceClient sc = nh_->serviceClient<cav_srvs::PlanManeuvers>(topic);
-                        service_clients_.emplace(topic, sc);
-                    }
+                    // if (capability == STRATEGIC_PLAN_CAPABILITY)
+                    // {
+                    //     ros::ServiceClient sc = nh_->serviceClient<cav_srvs::PlanManeuvers>(topic);
+                    //     service_clients_.emplace(topic, sc);
+                    // }
                 }
             }
         }
     }
     
     std::vector<std::string> CapabilitiesInterface::get_topics_for_capability(std::string query_string) const
     {
+
+        ros::ServiceClient sc_s = nh_->serviceClient<cav_srvs::PluginList>("/plugins/get_strategic_plugin_by_capability");
```

Comment:
I think it is better to make sc_c as a class variable and only initialize once. Otherwise it creates a new service client everytime this function is called.